components/ws-manager/pkg/manager/annotations.go

```diff
@@ -79,8 +81,17 @@ const (
 
 // markWorkspaceAsReady adds annotations to a workspace pod
 func (m *Manager) markWorkspace(ctx context.Context, workspaceID string, annotations ...*annotation) error {
+	// use custom backoff, as default one fails after 1.5s, this one will try for about 25s
+	// we want to try harder to remove or add annotation, as failure to remove "gitpod/never-ready" annotation
+	// would cause whole workspace to be marked as failed, hence the reason to try harder here.
+	var backoff = wait.Backoff{
+		Steps:    7,
+		Duration: 100 * time.Millisecond,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
 	// Retry on failure. Sometimes this doesn't work because of concurrent modification. The Kuberentes way is to just try again after waiting a bit.
-	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+	err := retry.RetryOnConflict(backoff, func() error {
```

Comment:
@sagor999 Lets try it. Could it be that the pod is cached by the informer and the reason for the failure is that we get a stale version on each retry?

Reply:
I don't think so, since `findWorkspacePod` pulls pod info on each retry.